Sum type in multi return results in C error

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.3 LTS (WSL 2)
Processor: 16 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
CC version: cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

getwd: /home/daniel/Projects/v
vmodules: /home/daniel/.vmodules
vroot: /home/daniel/Projects/v
vexe: /home/daniel/Projects/v/v
vexe mtime: 2021-12-21 17:33:15
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 5f0160b.f586c6b

Git version: git version 2.25.1
Git vroot status: weekly.2021.50.3-62-gabe3739f (11 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main/main.v`
```v
module main

type Abc = string | int | u32 | rune

fn cyz() (Abc, string) {
	return 'a', 'b'
}

fn main() {
	x, y := cyz()
	println(x)
	println(y)
}

```

**What did you expect to see?**

It should work as it is


**What did you see instead?**
```
==================
/tmp/v_1000/main.2768179633950027531.tmp.c:10897: error: cannot convert 'struct string' to 'struct string *'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4758,8 +4758,8 @@ fn (mut g Gen) return_stmt(node ast.Return) {
 			g.writeln('return $tmpvar;')
 			return
 		}
-		// typ_sym := g.table.sym(g.fn_decl.return_type)
-		// mr_info := typ_sym.info as ast.MultiReturn
+		typ_sym := g.table.sym(g.fn_decl.return_type)
+		mr_info := typ_sym.info as ast.MultiReturn
 		mut styp := ''
 		if fn_return_is_optional {
 			g.writeln('$ret_typ $tmpvar;')
@@ -4821,7 +4821,11 @@ fn (mut g Gen) return_stmt(node ast.Return) {
 			if expr.is_auto_deref_var() {
 				g.write('*')
 			}
-			g.expr(expr)
+			if g.table.sym(mr_info.types[i]).kind in [.sum_type, .interface_] {
+				g.expr_with_cast(expr, node.types[i], mr_info.types[i])
+			} else {
+				g.expr(expr)
+			}
 			arg_idx++
 			if i < node.exprs.len - 1 {
 				g.write(', ')

--- a/vlib/v/tests/multiret_with_sumtype_test.v
+++ b/vlib/v/tests/multiret_with_sumtype_test.v
@@ -1,0 +1,15 @@
+module main
+
+type Abc = int | rune | string | u32
+
+fn cyz() (Abc, string) {
+	return 'a', 'b'
+}
+
+fn test_multiret_with_sumtype() {
+	x, y := cyz()
+	println(x)
+	println(y)
+	assert x == Abc('a')
+	assert y == 'b'
+}

